cpp/core/shuffle/LocalPartitionWriter.cc

```diff
@@ -541,6 +541,10 @@ arrow::Status LocalPartitionWriter::evict(
   return arrow::Status::OK();
 }
 
+arrow::Status LocalPartitionWriter::evict(uint32_t partitionId, int64_t rawSize, const char* data, int64_t length) {
+  return arrow::Status::OK();
```

Comment:
Return `arrow::Status::NotImplemented("Invalid code path for local shuffle writer: sort based is not supported.");`